calblueprint/wecaresolar
# Instructions View

## 🖋 PR Description

Implemented and styled article view which displays the steps written in each article.  A progress bar is located at the top of the view which updates based how far the viewer has scrolled.

Notes:

- The steps of each article are stored under 'section' in Redux and with a label, title, imageUrl, and text field
- Each of the steps are rendered as card components, which automatically resized based on the content stored in the section
- Thank you @joyceyguo for the progress bar component!
- In order to implement the progress bar based on how far the viewer has scrolled, I utilized useState to update how the user has scrolled. Based on how far the user has scrolled and the dimensions of the display, a percentage is calculated

### 🤳Screenshots (if applicable)
<img width="529" alt="Screen Shot 2021-04-21 at 10 37 23 PM" src="https://user-images.githubusercontent.com/57604304/115661334-26b34400-a2f2-11eb-9020-197f95209f8d.png">
<img width="533" alt="Screen Shot 2021-04-21 at 10 37 53 PM" src="https://user-images.githubusercontent.com/57604304/115661389-37fc5080-a2f2-11eb-8546-996a2b11b114.png">

### 🐞 Bug Report

- The primary bug I ran into was the styling of the steps. I initially attempted to just separate each section using divs, however, I decided to switch to rendering each section as a card component. The card components automatically adjust based on the size of the content when using card media. 

- Another bug I ran into was rendering the images. I wasn't sure how to account for the images potentially being different orientations. In order to account for this, I decided to use the contain selection of objectFit and only specified a width for the image.

### 🧪 Testing

How do we test your feature?

Click on any of the article card to see the article view. Scrolling for views containing a scroll will update the progress bar. 
*Some views may not render any sections because the airtable hasn't been updated

cc: @susan-lin
